### DISCUSSION TITLE
Mapping an application in vehicles from the mapping_config.json file without using ["applications": "org.eclipse.mosaic.app.tutorial.cam.CamSendingApp"]

### DISCUSSION THREAD

**Discussion by Jean-Tshibangu-jtm** [eclipse/mosaic#373] [Q&A]
Hello,

I took the Java code from the CAM message application example, and generated a jar file with maven. And I put the jar file in the application folder of the scenario.

I share with you my mapping_config.json file : 

{
  "prototypes": [
      {
          "name": "tkrauss",
          "vehicleClass": "ElectricVehicle",
          "accel": 2.6,
          "decel": 4.5,
          "emergencyDecel": 6.0,
          "length": 5.00,
          "maxSpeed": 70.0,
          "minGap": 2.5,
          "sigma": 0.5,
          "tau": 1,
          "speedFactor": 1.0,
          "speedMode": "DEFAULT",
          "laneChangeMode": "DEFAULT",
          "deviations": 
          {
              "speedFactor": 0.0,
              "length": 0.0,
              "minGap": 0.0,
              "maxSpeed": 0.0,
              "accel": 0.0,
              "decel": 0.0,
              "tau": 0.0
          },
          "applications": ["org.eclipse.mosaic.app.tutorial.cam.CamSendingApp"],
         "weight": 0.7
      },
      {
        "name": "RSU",
        "applications": ["org.eclipse.mosaic.app.tutorial.RoadSideUnitApp"]
    }

      
  ],

  "servers": [
      {
          "name": "WeatherServer",
          "group": "WeatherServer",
          "applications": [ "org.eclipse.mosaic.app.tutorial.WeatherServerApp" ]
      }
      
  ],
  
 
  "rsus": [
    {
        "name": "RSU",
        "position": {
            "latitude": 48.827350,
            "longitude": 1.954738
        }
    },
    { 
        "name": "RSU",
        "position": {
            "latitude": 48.834734, 
            "longitude": 1.957318
        }
    }
 ]
}

Questions: 
1. How do I map my application to vehicles from mapping_config.json without using
["applications": "org.eclipse.mosaic.app.tutorial.cam.CamSendingApp"],

2. Follow this link https://eclipse.dev/mosaic/docs/simulators/application_simulator/, 
How to configure the file ( your_application_config.json);

**Comment by schwepmo** [comment]
Dear @Jean-Tshibangu-jtm,
sorry for getting back so late, we were busy/out of office during the holidays.
Regarding your questions:
1. you can also add a list of applications into your defined `prototypes` in the same manner, by adding a the respective list `"applications": ["your.app.Class"]`. However, you are bound to the *mapping_config.json*.
2. https://github.com/eclipse/mosaic/tree/main/app/tutorials/example-applications/src/main/java/org/eclipse/mosaic/app/tutorial/configurableapp - here you can find an example application that extends `ConfigurableApplication`. Your own apps would also have to extend `ConfigurableApplication` to access this functionality

I hope this helps you, feel free to further extend your questions.
